syntax = "proto3";

package transaction_verification;

service VerificationService {
    rpc TransactionVerification (VerificationRequest) returns (VerificationResponse);
    rpc UpdateVectorClock (VectorClockRequest) returns (Empty);
}

message VerificationRequest {
    string user_name = 1;
    string user_contact = 2;
    string creditcard_nr = 3;
    repeated string items = 4;
    repeated int32 quantities = 5;
    string orderId = 6;
}

message VerificationResponse {
    bool isOk = 1;
}

message VectorClockRequest {
    string orderId = 1;
    repeated int32 vectorClock = 2;
    bool shouldCancel = 3;
}

message Empty {}
